messenger/http_transporter.go

```diff
@@ -55,30 +57,88 @@ func NewHTTPTransporter(upid *upid.UPID) *HTTPTransporter {
 	}
 }
 
+// some network errors are probably recoverable, attempt to determine that here.
+func isRecoverableError(err error) bool {
+	if urlErr, ok := err.(*url.Error); ok {
+		log.V(2).Infof("checking url.Error for recoverability")
+		return urlErr.Op == "Post" && isRecoverableError(urlErr.Err)
+	} else if netErr, ok := err.(*net.OpError); ok && netErr.Err != nil {
+		log.V(2).Infof("checking net.OpError for recoverability: %#v", err)
+		if netErr.Temporary() {
+			return true
+		}
+		//TODO(jdef) this is pretty hackish, there's probably a better way
+		return (netErr.Op == "dial" && netErr.Net == "tcp" && strings.HasSuffix(netErr.Error(), ": connection refused"))
```

Comment:
I could probably test the captured error against syscall.ECONNREFUSED too

On Tue, Feb 10, 2015 at 12:39 PM, James DeFelice <james.defelice@gmail.com>
wrote:

> I was trying to make a more-specific constraint vs. a less-specific one,
> with the intent to avoid matching against too many things. Because this is
> a hacked approach, I want to minimize potential side-effects. As it
> currently stands, there's no need to test "tcp" but doing so would be less
> specific.
>
> On Tue, Feb 10, 2015 at 12:24 PM, Vladimir Vivien <
> notifications@github.com> wrote:
>
>> In messenger/http_transporter.go
>> <https://github.com/mesos/mesos-go/pull/70#discussion_r24430919>:
>>
>> > @@ -55,30 +57,88 @@ func NewHTTPTransporter(upid *upid.UPID) *HTTPTransporter {
>> >  	}
>> >  }
>> >
>> > +// some network errors are probably recoverable, attempt to determine that here.
>> > +func isRecoverableError(err error) bool {
>> > +	if urlErr, ok := err.(*url.Error); ok {
>> > +		log.V(2).Infof("checking url.Error for recoverability")
>> > +		return urlErr.Op == "Post" && isRecoverableError(urlErr.Err)
>> > +	} else if netErr, ok := err.(*net.OpError); ok && netErr.Err != nil {
>> > +		log.V(2).Infof("checking net.OpError for recoverability: %#v", err)
>> > +		if netErr.Temporary() {
>> > +			return true
>> > +		}
>> > +		//TODO(jdef) this is pretty hackish, there's probably a better way
>> > +		return (netErr.Op == "dial" && netErr.Net == "tcp" && strings.HasSuffix(netErr.Error(), ": connection refused"))
>>
>> Is there a need to test for "tcp" ? Also, maybe strings.Contains() for
>> future proof (but, still hackish) ?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mesos/mesos-go/pull/70/files#r24430919>.
>>
>
>
>
> --
> James DeFelice
> 585.241.9488 (voice)
> 650.649.6071 (fax)
>



-- 
James DeFelice
585.241.9488 (voice)
650.649.6071 (fax)